email.go

```diff
@@ -211,31 +221,38 @@ func (e *Email) Send(addr string, a smtp.Auth, TLSMandatory bool) error {
 		}
 		to[i] = addr
 	}
+
 	// Check to make sure there is at least one recipient and one "From" address
 	if e.From == "" || len(to) == 0 {
 		return errors.New("Must specify at least one From address and one To address")
 	}
 
 	from, err := extractEmail(e.From)
-	if err != nil {
-		return err
-	}
 
 	msg, err := e.Bytes()
 	if err != nil {
 		return err
 	}
-	return sendMail(addr, a, from, to, msg, TLSMandatory)
-}
 
-var testHookStartTLS func(*tls.Config) // nil, except for tests
+	return sendMailUsingConn(conn, addr, from, to, msg, TLSMandatory, a)
+}
 
 // Copy of smtp.SendMail() modified so we can fail if STARTTLS isn't available.
-func sendMail(addr string, a smtp.Auth, from string, to []string, msg []byte, TLSMandatory bool) error {
-	c, err := smtp.Dial(addr)
+func sendMailUsingConn(
+	conn net.Conn,
+	addr string,
+	from string,
+	to []string,
+	msg []byte,
+	TLSMandatory bool,
+	a smtp.Auth,
+) error {
+	host, _, _ := net.SplitHostPort(addr)
```

Comment:
Are we guaranteed that the address is a valid format right here?  If not, we should check the error.

Reply:
See line 56 of stdlib, I'm just copying their logic.

https://golang.org/src/net/smtp/smtp.go#56

Reply:
stdlib is bad.

Reply:
I guess the next line would fail anyways if this fails.